opentracing/tracer.py

```diff
@@ -38,33 +40,88 @@ class Tracer(object):
     _supported_formats = [Format.TEXT_MAP, Format.BINARY, Format.HTTP_HEADERS]
 
     def __init__(self):
+        self._scope_manager = ScopeManager()
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._noop_scope = Scope(self._scope_manager, self._noop_span)
+
+    @property
+    def scope_manager(self):
+        """ScopeManager accessor"""
+        return self._scope_manager
+
+    def start_active(self,
```

Comment:
nomenclature nitpick: i don't like the naming of `start_active` vs `start_manual`. `start_active` will be used 95% of the time, so i think it should just be called `start` and then there can be `start_{whatever_you_choose}` for the rest.

Reply:
that's an interesting point since the most common API may be the shortest possible, let's see what others are thinking!

Reply:
@carlosalberto thoughts?

Reply:
I would vote to keep the api as close to the other language implementations as possible. On a similar note, should something about in process span management be added to the opentracing specification?

Reply:
If I'm understanding the [Java API correctly](http://javadoc.io/doc/io.opentracing/opentracing-api/0.31.0-RC2), it looks like `Tracer.buildSpan(...).start() == Tracer.buildSpan(...).start_manual()`. So, if those conventions would be followed, would `Tracer.start_span() == Tracer.start_manual()` in the Python API?

If `start_active()` is the reasonable default, as @clutchski mentions, it feels a bit unfortunate. IMO, most developers instrumenting their code will begin with `Tracer.start_span()` as they might not want to deal with the cognitive overhead of understanding the difference btw `active` and `manual` immediately.

Either way, I'd be for:

* not deprecating `start_span` so there's one less decision when adding your first span as a developer
* adding a real-world example to the docs on when to use `active` vs. `manual`. It feels a bit abstract for a developer that isn't writing the OpenTracing API implementations.

Reply:
I think that following the Java api we should only have `start` and `start_active` (that being said, I understand there are pros and cons for both mentioned naming options).

`start_span` is not part the official API yet so we could get rid of it now. 

About documentation: It feels we should add some general info now, as opposed to waiting for the final release, so people trying it out know what to expect.

And about more advanced examples, I started putting a few examples here: https://github.com/carlosalberto/python-examples (that mimic the ones in Java). I'm half done with them using the latest `Scope` concept, and I hope we will make them part of the main repo eventually.

Reply:
yes, in most languages the API do not use the notion of span builder, therefore the main method is called `startSpan` / `start_span`. `start_active_span` makes sense.

Reply:
I agree, lets keep the `_span` suffix (but lets leave `start_manual_span` out, as having an alias it's not a good idea IHMO )

Reply:
+1, "manual" was removed from the Java API.

Even though `start_active_span` will be used more often, it's worth keeping the full name to be in sync with other languages and emphasize the side effects. It also doesn't return a Span the way `start_span` does.

Reply:
> It also doesn't return a Span the way start_span does.

Ah - I overlooked that. Seems like it should be`Tracer.start_active_scope()`.

I've submitted a PR to @palazzem's branch with these naming updates (https://github.com/palazzem/opentracing-python/pull/1).

Reply:
I think it should be `start_active_span`, since Span is the main thing that's being created here, it's just wrapped in a Scope for management.

Reply:
I don't have a strong opinion on `_active_`, so I'm happy to adjust it. 

Here's the difference in practice:

```
with tracer.start_active_span('...') as scope:
                scope.span().set_tag('http.method', 'GET')
                do_some_work()
```

vs.

```
with tracer.start_active_scope('...') as scope:
```

Reply:
Sure thing! I definitely think that `start_manual*` must be out of the Python API. Let's make this change.